docs/src/main/asciidoc/lra-guide.adoc

```diff
@@ -0,0 +1,196 @@
+         * invoked.
+         */
+    }
+
+    @Complete
+    @Path("/complete")
+    @PUT
+    public Response completeWork(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId,
+                                 String userData)
+    {
+        /*
+         * Free up resources allocated in the context of the LRA identified by the
+         * value in the injected JAX-RS header.
+         *
+         * Since there is no @Status method in this class, completeWork MUST be
+         * idempotent and MUST return the status.
+         */
+         return Response.ok(ParticipantStatus.Completed.name()).build();
+    }
+
+    @Compensate
+    @Path("/compensate")
+    @PUT
+    public Response compensateWork(@HeaderParam(LRA_HTTP_CONTEXT_HEADER) URI lraId,
+                                   String userData)
+    {
+        /*
+         * The LRA identified by the value in the injected JAX-RS header was
+         * cancelled so the business logic should compensate for any actions
+         * that have been performed while running in its context.
+         *
+         * Since there is no @Status method in this class, compensateWork MUST be
+         * idempotent and MUST return the status
+         */
+         return Response.ok(ParticipantStatus.Compensated.name()).build();
+    }
+}
+----
+
+The example also shows that when an LRA is present its identifier can be obtained
+by reading the request headers.
+
+The same section of the specification document also includes an
+example that demonstrates how to start an LRA in one method and close
+it in a different resource method using the `LRA#end` element.
+It also shows how to configure the LRA to be automatically cancelled if the business method
+returns the particular HTTP status codes identified in the `cancelOn` and
+`cancelOnFamily` elements:
+
+[source,java]
+----
+  @LRA(value = LRA.Type.REQUIRED, // if there is no incoming context a new one is created
+       cancelOn = {
+           Response.Status.INTERNAL_SERVER_ERROR // cancel on a 500 code
+       },
+       cancelOnFamily = {
+           Response.Status.Family.CLIENT_ERROR // cancel on any 4xx code
+       },
+       end = false) // the LRA will continue to run when the method finishes
+  @Path("/book")
+  @POST
+  public Response bookTrip(...) { ... }
+
+  @LRA(LRA.Type.MANDATORY, // requires an active context before method can be executed
+       end = true) // end the LRA started by the bookTrip method
+  @Path("/confirm")
+  @PUT
+  public Booking confirmTrip(Booking booking) throws BookingException { ... }
+----
+
+The `end = false` element on the bookTrip method forces the LRA to continue running when
+the method finishes and the `end = true` element on the confirmTrip method forces the LRA
+(started by the bookTrip method) to close the LRA. Note that this end element can
+be placed on any JAX-RS resource (ie one service can start the LRA whilst a different
+service ends it).
```

Comment:
No, the guide is to help users get started with the extension.
There is only one configuration option and it does not need a specific section for it. The text that refers to the option flows just fine with the section where I placed it.

Reply:
ok

Reply:
It's very interesting, but we may want to refactor this document into:

1. Getting started with LRA - just plain getting started, nothing else 
2. An LRA reference guide with these examples

All this content is great, but we also need to distillate it at the right pace for users that are not in the LRA world. It's a difficult exercise, and I/we failed multiple times.

Reply:
So when there is some guidance I will split it up.
I intentionally made the guide minimal since there is content elsewhere and I was keen to avoid duplication and the maintenance impact that brings.

Reply:
Here are some rules I'm trying to apply:

* getting started should only contain details to run an example with a small introduction. It should not expect any prior knowledge, and from the beginning to the end should not take more than 15 minutes. 
* reference contains everything else, more complex example, configuration.... 

So, typically, from line 164, it should be in the reference guide. 
Note that of course, you can cross-link.

Reply:
The guide is already minimal, so you are requesting that I reduce it even more?

Reply:
The last part looks more advanced. At least it's how I felt it when I read it.

Reply:
We are doing microservices.